crates/nu-engine/src/documentation.rs

```diff
@@ -27,70 +32,223 @@ pub fn get_full_help(
         engine_state,
         stack,
-        command.is_keyword(),
+        &(&call).into(),
+        Value::string(code_string, Span::unknown()).into_pipeline_data(),
     )
+    .and_then(|pipe| pipe.into_value(Span::unknown()))
+    .and_then(|val| val.coerce_into_string())
+    .ok()
 }
 
-/// Syntax highlight code using the `nu-highlight` command if available
+/// Syntax highlight code using the `nu-highlight` command if available, falling back to the given string
 fn nu_highlight_string(code_string: &str, engine_state: &EngineState, stack: &mut Stack) -> String {
-    if let Some(highlighter) = engine_state.find_decl(b"nu-highlight", &[]) {
-        let decl = engine_state.get_decl(highlighter);
+    try_nu_highlight(code_string, false, engine_state, stack)
+        .unwrap_or_else(|| code_string.to_string())
+}
 
-        let call = Call::new(Span::unknown());
+/// Apply code highlighting to code in a capture group
+fn highlight_capture_group(
+    captures: &Captures,
+    engine_state: &EngineState,
+    stack: &mut Stack,
+) -> String {
+    let Some(content) = captures.get(1) else {
+        // this shouldn't happen
+        return String::new();
+    };
 
-        if let Ok(output) = decl.run(
-            engine_state,
-            stack,
-            &(&call).into(),
-            Value::string(code_string, Span::unknown()).into_pipeline_data(),
-        ) {
-            let result = output.into_value(Span::unknown());
-            if let Ok(s) = result.and_then(Value::coerce_into_string) {
-                return s; // successfully highlighted string
-            }
-        }
+    // Save current color config
+    let config_old = stack.get_config(engine_state);
+    let mut config = (*config_old).clone();
+
+    // Style externals and external arguments with fallback style,
+    // so nu-highlight styles code which is technically valid syntax,
+    // but not an internal command is highlighted with the fallback style
+    let code_style = Value::record(
+        record! {
+            "attr" => Value::string("di", Span::unknown()),
+        },
+        Span::unknown(),
+    );
+    let color_config = &mut config.color_config;
+    color_config.insert("shape_external".into(), code_style.clone());
+    color_config.insert("shape_external_resolved".into(), code_style.clone());
+    color_config.insert("shape_externalarg".into(), code_style);
+
+    // Apply config with external argument style
+    stack.config = Some(Arc::new(config));
+
+    // Highlight and reject invalid syntax
+    let highlighted = try_nu_highlight(content.into(), true, engine_state, stack)
+        // Make highlighted string italic
+        .map(|text| {
+            // replace resets with reset + italic, so the whole string is actually italicized
+            let text = text.replace(RESET, &format!("{RESET}{DEFAULT_ITALIC}"));
+            // start italicized, and also undo the final reset -> reset+italic
+            format!("{DEFAULT_ITALIC}{text}{RESET}")
```

Comment:
If I'm reading this right and `text` was `fdncred{RESET}` it would be changed to `fdncred{RESET}{DEFAULT_ITALIC}` and then this would change it to `{DEFAULT_ITALIC}fdncred{RESET}{DEFAULT_ITALIC}{RESET}`. I'm not sure all that is needed (or if I'm interpreting it right) but I'd have to see the real escapes to be sure.

Reply:
yea that's correct, the final `{RESET}{DEFAULT_ITALIC}` isn't needed so this is kind of a hack (which I can polish up if we decide to go this direction). it is necessary to insert the italic code after each reset though because `nu-highlight` has to insert a reset after each new style is applied (you can see here there's a reset after `where` for example)

![image](https://github.com/user-attachments/assets/80b71afa-2a36-42c5-9aa5-10e480b57d8f)